cadl-extension/src/main/java/com/azure/autorest/CadlPlugin.java

```diff
@@ -0,0 +1,165 @@
+            writer.write(content);
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
+        }
+        LOGGER.info("Write file: {}", fileName);
+    }
+
+    private static boolean isGeneratingModel(ClientModel model) {
+        return model.getImplementationDetails() != null
+                && (model.getImplementationDetails().isConvenienceMethod() || JavaSettings.getInstance().isGenerateModels())
+                && !model.getImplementationDetails().isException();
+    }
+
+    private static boolean isGeneratingModel(EnumType model) {
+        return model.getImplementationDetails() != null
+                && (model.getImplementationDetails().isConvenienceMethod() || JavaSettings.getInstance().isGenerateModels())
+                && !model.getImplementationDetails().isException();
+    }
+
+    private static boolean isGeneratingModel(ClientResponse response) {
+        IType bodyType = response.getBodyType();
+        boolean ret = ClientModelUtil.isClientModel(bodyType);
+        if (ret) {
+            ClassType classType = (ClassType) bodyType;
+            ClientModel model = ClientModelUtil.getClientModel(classType.getName());
+            if (model != null) {
+                ret = isGeneratingModel(model);
+            }
+        }
+        return ret;
+    }
+
+    private static final Map<String, Object> SETTINGS_MAP = new HashMap<>();
+    static {
+        SETTINGS_MAP.put("data-plane", true);
+
+        SETTINGS_MAP.put("license-header", "MICROSOFT_MIT_SMALL");
+        SETTINGS_MAP.put("generate-client-interfaces", false);
+        SETTINGS_MAP.put("generate-client-as-impl", true);
+        SETTINGS_MAP.put("generate-sync-async-clients", true);
+        SETTINGS_MAP.put("generate-builder-per-client", true);
+        SETTINGS_MAP.put("add-context-parameter", true);
+        SETTINGS_MAP.put("context-client-method-parameter", true);
+        SETTINGS_MAP.put("sync-methods", "all");
+
+        SETTINGS_MAP.put("use-default-http-status-code-to-exception-type-mapping", true);
+        SETTINGS_MAP.put("polling", new HashMap<String, Object>());
+
+        SETTINGS_MAP.put("models-subpackage", "models");
+        SETTINGS_MAP.put("client-logger", true);
+        SETTINGS_MAP.put("model-override-setter-from-superclass", true);
+        SETTINGS_MAP.put("custom-strongly-typed-header-deserialization", true);
+        SETTINGS_MAP.put("required-fields-as-ctor-args", true);
+
+        SETTINGS_MAP.put("regenerate-pom", true);
+        SETTINGS_MAP.put("generate-models", true);
+    }
+
+    public static class MockConnection extends Connection {
+
+        public MockConnection() {
+            super(null, null);
+        }
+    }
+
+    public CadlPlugin(String namespace) {
+        super(new MockConnection(), "dummy", "dummy");
+        SETTINGS_MAP.put("namespace", namespace);
+        JavaSettingsAccessor.setHost(this);
+        JavaSettings.getInstance();
```

Comment:
Is this line doing anything? It's a getter with the return being used

Reply:
Likely no. Will try remove it.